utils/unittest/fixtures.go

```diff
@@ -1146,3 +1146,43 @@ func ChainFixtureFrom(count int, parent *flow.Header) []*flow.Block {
 
 	return blocks
 }
+
+func ReceiptChainFor(blocks []*flow.Block, result0 *flow.ExecutionResult) []*flow.ExecutionReceipt {
+	receipts := make([]*flow.ExecutionReceipt, len(blocks))
+	receipts[0] = ExecutionReceiptFixture(WithResult(result0))
+	receipts[0].ExecutionResult.BlockID = blocks[0].ID()
+
+	for i := 1; i < len(blocks); i++ {
+		b := blocks[i]
+		prevReceipt := receipts[i-1]
+		receipt := ReceiptForBlockFixture(b)
+		receipt.ExecutionResult.PreviousResultID = prevReceipt.ExecutionResult.ID()
+		prevLastChunk := prevReceipt.ExecutionResult.Chunks[len(prevReceipt.ExecutionResult.Chunks)-1]
+		receipt.ExecutionResult.Chunks[0].StartState = prevLastChunk.EndState
+		receipts[i] = receipt
+	}
+
+	return receipts
+}
+
+// ReconnectBlocksAndReceipts re-computes each block's PayloadHash and ParentID
+// so that all the blocks are connected.
+// blocks' height have to be in strict increasing order.
+func ReconnectBlocksAndReceipts(blocks []*flow.Block, receipts []*flow.ExecutionReceipt) {
```

Comment:
a useful helper function that in the future we could extend to create multiple execution chains for the same chain of blocks.